scripts/merge_tables.R

```diff
@@ -92,19 +108,80 @@ if ( is.null(in.dir) ) {
 ###    FUNCTIONS   ###
 ######################
 
-merge_tables <- function(cwd, prefix){
-  dataFiles <- dir(cwd, prefix, full.names=TRUE)
+#' Read and process input table
+#'
+#' `get_table()` uses `tryCatch()` to read the file in `tbl_pth`. If the table
+#' is empty and an error is raised, a data frame is created.
+#'
+#' @param tbl_pth Path to the input table.
+#' @param prefix String to be removed from the input file name. It must be
+#'   present in all the tables to be merged.
+#'
+#' @returns `get_table()` returns a data frame containing the miRNA species to
+#'   be counted in first column, named `ID`, and their counts in that file in
+#'   the second one. The name of the second column in the data frame is obtained
+#'   by removing the `prefix` from the input file name. If no `prefix` is given,
+#'   the whole file name is used.
+#'
+#'   If the input file is empty, the returned data frame will consist on one row
+#'   with a `NA` in both fields.
+#'
+#' @seealso [tryCatch()] which this function uses.
+get_table <- function(tbl_pth, prefix) {
+  sample <- gsub(prefix, "", tbl_pth)
+  fields <- c("ID", basename(sample))
+  
+  tryCatch(
+    expr = {
+      table <- read.table(tbl_pth, sep = '\t', col.names = fields)
+      return(table)
+    },
+    error = function(e) {
+      table <- data.frame(matrix(NA, ncol = 2, nrow = 1))
+      colnames(table) <- fields
+      return(table)
+    }
+  )
+}
+
+#' Merge tables with the same prefix
+#'
+#' `merge_tables()` takes all the files in `cwd` that start with `prefix` and
+#' merge them keeping all the miRNA species present in each of the tables.
+#'
+#' @details The function `get_table()` is used to make sure that even if an
+#'   empty input file is given, the merge can still be done. Thus, before
+#'   returning the merged table, the row with a `NA` in the `ID` field, if any,
+#'   is removed.
+#'
+#'   The function `dplyr::full_join()` method is used for the merge, therefore,
+#'   if a miRNA species in `ID` is missing in any of the tables being joined,
+#'   its value is set to `NA` in that column.
```

Comment:
I think this description is a bit confused/confusing. It would probably make sense to first describe why and under what conditions rows with `NA` in the `ID` field are added before you say that they are removed. Also, both the `Thus, ` and `therefore, ` seem to not describe something that is logically following (at least not immediately/obviously), so I guess some other words should be used here.